### PR TITLE
back port changes for issue 7112 to v1.13

### PR DESCRIPTION
**Pull request by sheidkamp**
# Description
Backport of https://github.com/solo-io/gloo/pull/7661 to v1.13
for issue
https://github.com/solo-io/gloo/issues/7112
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/7112

### CODE REVIEW

File: changelog/v1.13.4/expose-kube-jobspec-fields.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+changelog:
+  - type: FIX
```
Comment (sam-heilbron) [inline changelog/v1.13.4/expose-kube-jobspec-fields.yaml:2]: Let's make this a helm change

File: changelog/v1.13.4/expose-kube-jobspec-fields.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+    description: Backport - Add backoffLimit, completions, manualSelector, and parallelism fields to k8s jobs. https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#job-v1-batch
```
Comment (sam-heilbron) [inline changelog/v1.13.4/expose-kube-jobspec-fields.yaml:3]: I don't think we need to include the "backport" reference in the changelog. These are aggregated and published to our docs, so people viewing them mainly care about the fix/features introduced.

**Review by jenshu — APPROVED** [review]

**Review by sam-heilbron — APPROVED** [review]

**Pull request merged**